### ISSUE TITLE
False positive for trailing_closure

### ISSUE THREAD

**Issue by atkinsa1** [realm/SwiftLint#3563]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
When calling a method with multiple optional closure arguments, trailing_closure triggers when you specify one such argument.
This is prevalent with the Combine method handleEvents.
```swift
myPublisher
  .handleEvents(receiveOutput: {
    // do something
  })
  .store(in: &cancellables)
```

All arguments for handleEvents are optional closures.
You cannot use a trailing closure here because swift cannot determine which argument you are passing.

### Environment

* SwiftLint version 0.43.0
* Cocoapods

**Comment by atkinsa1** [comment]
Sorry, saw this was mitigated with:
```
trailing_closure:
  only_single_muted_parameter: true
```
